core/src/main/java/org/opensearch/remote/metadata/client/AbstractSdkClient.java

```diff
@@ -139,6 +139,10 @@ protected CompletionStage<GetDataObjectResponse> addToGlobalResourceCache(
         CompletionStage<GetDataObjectResponse> dataFetchedWithGlobalTenantId
     ) {
         return dataFetchedWithGlobalTenantId.thenCompose(res -> {
+            GetResponse getResponse = res.getResponse();
+            if (getResponse == null || !getResponse.isExists()) {
+                return CompletableFuture.completedFuture(res);
+            }
```

Comment:
While this solves the immediate problem of returning the "not exists" to the caller, it seems we're missing the chance here to clear the cache.   If TTL is a very long value (or 0) this will never happen.

Reply:
> Currently we depends on TTL to clear cache.

Ah, I see.  In the normal (sdkClient.get) flow, if we are past TTL the cache is already cleared before we get here; if we have a very long (or 0) TTL, we always return from the cache and never get here.

So it's sufficient here to just not insert into the cache.

Reply:
Yes, one minor thing is 0 means no cache at all in current implementation.